### DISCUSSION TITLE
delete column using TableMigration

### DISCUSSION THREAD

**Discussion by ahuruFoundation** [simolus3/drift#1089] [Q&A]
Is there an example of how to delete column using TableMigration? I have look around a bit and could not see an example about this. Any help is appreciated.

**Comment by kuhnroyal** [comment]
This requires a complex migration: https://moor.simonbinder.eu/docs/advanced-features/migrations/#complex-migrations

**Comment by simolus3** [comment]
As Peter said, you need a complex migration. If you don't have any foreign key constraints pointing _towards_ that column, deleting it is actually very simple:

```dart
await m.alterTable(TableMigration(yourTable));
```

Basically, `alterTable` will create a new table with a different name, copy all the data over, delete the old table and finally rename the new table. So if you want to delete a column, no special transformation is necessary. I'll add this example to the documentation as well.

**Reply by ahuruFoundation** [reply]
@simolus3 That's exactly what I need. Thanks!
